File: src/main/java/com/xeoneux/caravansary/repository/PageableRoomRepository.java
package com.xeoneux.caravansary.repository;

import com.xeoneux.caravansary.entity.RoomEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;

public interface PageableRoomRepository extends PagingAndSortingRepository<RoomEntity, Long> {
    Page<RoomEntity> findById(Long id, Pageable page);
}

File: src/main/kotlin/com/xeoneux/caravansary/repository/PageableRoomRepository.kt
package com.xeoneux.caravansary.repository

import com.xeoneux.caravansary.entity.RoomEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository

interface PageableRoomRepository : PagingAndSortingRepository<RoomEntity, Long> {
    fun findById(id: Long?, page: Pageable): Page<RoomEntity>
}
